apixu-weatherbot1.0.tcl

```diff
@@ -0,0 +1,498 @@
+                    # Converts date to the day of the week.
+                    set dayname [clock format [clock scan $date] -format %a]
+                    set condition [dict get $forecast day condition text]
+                    set maxtemp_c [dict get $forecast day maxtemp_c]
+                    set mintemp_c [dict get $forecast day mintemp_c]
+                    set maxtemp_f [dict get $forecast day maxtemp_f]
+                    set mintemp_f [dict get $forecast day mintemp_f]
+
+                    switch $units {
+                        "0" {
+                            append spam " - \002$dayname:\002 $condition (High: $maxtemp_c\C\
+                                         Low: $mintemp_c\C)"
+                            continue
+                        }
+                        "1" {
+                            append spam " - \002$dayname:\002 $condition (High: $maxtemp_f\F\
+                                         Low: $mintemp_f\F)"
+                            continue
+                        }
+                        "2" {
+                            append spam " - \002$dayname:\002 $condition (High: $maxtemp_f\F/$maxtemp_c\C\
+                                         Low: $mintemp_f\F/$mintemp_c\C)"
+                            continue
+                        }
+                    }
+                }
+
+                return $spam
+            }
+        }
+    }
+
+    proc _set_location {nick uhost hand text} {
+        # Sets a users location and units in the eggdrop userfile.
+        #
+        # Args:
+        #   nick(string) - Users nick.
+        #   ushost(string) - Users username and host.
+        #   hand(string) - Users handle.
+        #   text(string) - Units and location to be set.
+        #
+        # Returns:
+        #   (dict) - Returns the userinfo(units and location)
+        #            that was set to the bot.
+        #
+        putlog "weather::location was called"
+        putlog "nick: $nick, uhost: $uhost, hand: $hand is trying to set location"
+        set units [string index $text 0]
+        set location [string range $text 2 end]
+
+        # First regexp matches any location given by user that is not
+        # a-z, A-Z, number, period, comma or space. This is sanitize
+        # the location string given to catch any illegal characters
+        # before the API call that will return an error.
+        #
+        # Second regexp matches any location give by user that starts
+        # and ends with a-z or A-Z with a : in the between the two. This
+        # is to let strings like "iata:ksjc" to be passed through as the
+        # only illegal character if used the correct way.
+        #
+        # If the first regexp is true and second regexp is false, the
+        # conditional will be true and throw an error for illegal characters.
+        if {[regexp {[^\w., ]} $location] && ![regexp {^[a-zA-Z]+:[a-zA-Z]+$} $location]} {
```

Comment:
hai this looks familiar, could clean it up with making that parse location function idea mentioned above